### PR TITLE
feat : Added Wishlist and Cart

### PR DESCRIPTION
**Pull request by Bharatrathod-81**
Added cart page and wishlist page 
Used useReducer  and Context

### CODE REVIEW

File: src/components/cart-listing/cart-listing.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                                <img className="cart-card-Img" src={item.image} />
+                                <div className="cart-card-text">
+                                    <h4 className="cart-car">{item.categoryName}</h4>
+                                    <small className="cart-car">{item.model}</small>
+                                    <h3>₹{item.price}<span className="max-price"> ₹{ item.price*2}</span></h3>
+                                    <span className="discount-text">50% off</span>
+                                    <div className="product-quantity">
+                                        <small>Quantity</small>
+                                        <button
+                                         onClick={() => {
```
Comment (tanishk15sharma) [inline src/components/cart-listing/cart-listing.jsx]: No need to use curly braces `{}` here .

**Review by tanishk15sharma — APPROVED** [review]
You can use a code editor like `prettier` for better readability . in some places you are using `{}` even for single line of code you can correct it . 
I liked that your reducer file is separate .

File: src/components/cart-listing/cart-listing.css [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+.max-price{
+    color: gray;
+    font-size: larger;
+    text-decoration :line-through;
+}
+
+.discount-text{
+    color: gray;
```
Comment (nikhilmalviya-dev) [inline src/components/cart-listing/cart-listing.css:63]: you can use global variable for repeating color....

**Review by nikhilmalviya-dev — APPROVED** [review]

**Review by Omkar-Chaskar — COMMENTED** [review]
code is clean and well written.
components are well structured.
Just Simple suggestion , In css file flex property in been used many times.
You can initiate use new class which contain flex property as we done in our Component Library for Code reduction.

File: src/components/single-product/single-product.jsx [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+        wishlist, cart
+        }, dispatchWishlistCart } = useWishlistCartContext();
+
 
     return(
         <div className="cards">
             <div>
-            <img className="card-Img" src={data.image}/>
-            <button className="heart-Btn">&#10084;</button>
+                <img className="card-Img" src={data.image}/>
```
Comment (ashishnnnnn) [inline src/components/single-product/single-product.jsx]: These images could have alt attribute..

**Review by ashishnnnnn — CHANGES_REQUESTED** [review]

**Pull request merged**